custom/ewsghana/balance.py

```diff
@@ -243,3 +246,94 @@ def balance_migration(self, date=None):
         self.validate_supply_points(date)
         self.validate_web_users(date)
         self.validate_sms_users()
+
+    def _check_report(self, report, reports, day, interval):
+        if report.report not in EmailSettingsSync.REPORT_MAP or not report.users:
+            return False
+        username = report.users[0].lower()
+        location_code = report.view_args.split()[1][1:-2]
+        desc = "Report not migrated; "
+        report_tuple = (username, day, report.hours, location_code,
+                        EmailSettingsSync.REPORT_MAP[report.report], interval)
+        external_id = '{}-{}-{}-{}-{}-{}'.format(*report_tuple)
+        if report_tuple not in reports:
+            web_user = WebUser.get_by_username(username)
+            if not web_user or self.domain not in web_user.domains:
+                desc += 'User {} is not active; '.format(username)
+            try:
+                sql_location = SQLLocation.objects.get(domain=self.domain, site_code=location_code)
+                if sql_location.is_archived:
+                    desc += 'Location {} is not active; '.format(location_code)
+            except SQLLocation.DoesNotExist:
+                desc += 'Location {} does not exist; '.format(location_code)
+
+            migration_problem, _ = EWSMigrationProblem.objects.get_or_create(
+                domain=self.domain,
+                object_id=username,
+                object_type='email_report',
+                external_id=external_id
```

Comment:
I just realized, external_id is defined as varchar(128) - will the external_id ever exceed 128 characters or should it always fit?